packages/vc-verification/src/issuer-verification.ts

```diff
@@ -0,0 +1,221 @@
+import * as jwt from 'jsonwebtoken';
+import { providers } from 'ethers';
+import { ProofVerifier } from '@ew-did-registry/claims';
+import { Resolver, addressOf } from '@ew-did-registry/did-ethr-resolver';
+import { RegistrySettings } from '@ew-did-registry/did-resolver-interface';
+import { CredentialResolver, IssuerDefinitionResolver } from '.';
+import {
+  IVerifiableCredential,
+  VerificationResult,
+  OffChainClaim,
+} from './models';
+import { upgradeChainId } from './upgrade-chainid';
+
+export class IssuerVerification {
+  private _resolver: Resolver;
+  private _issuerDefResolver: IssuerDefinitionResolver;
+  private _credentialResolver: CredentialResolver;
+
+  /**
+   *
+   * @param provider
+   * @param registrySetting
+   * @param credentialResolver
+   * @param issuerDefResolver
+   */
+  constructor(
+    provider: providers.Provider,
+    registrySetting: RegistrySettings,
+    credentialResolver: CredentialResolver,
+    issuerDefResolver: IssuerDefinitionResolver
+  ) {
+    this._issuerDefResolver = issuerDefResolver;
+    this._resolver = new Resolver(provider, registrySetting);
+    this._credentialResolver = credentialResolver;
+  }
+
+  /**
+   * Verifies chain of trust for a holder's credential
+   * @param credential
+   * @param verifyCredentialProofCallback
+   * @returns
+   */
+  async verifyChainOfTrust(
+    credential: IVerifiableCredential,
+    verifyCredentialProofCallback?: (
+      credential: IVerifiableCredential
+    ) => Promise<VerificationResult>
+  ) {
+    if (credential && verifyCredentialProofCallback) {
+      return this.verifyChainOfTrustCallback(
+        credential,
+        verifyCredentialProofCallback
+      );
+    } else if (credential) {
+      return this.verifyCredentialChainOfTrust(credential);
+    } else {
+      throw new Error('Not sufficient information');
+    }
+  }
+
+  /**
+   * Verifies chain of trust for a given holder's DID and role
+   * @param {string} credential
+   */
+  async verifyCredentialChainOfTrust(credential: IVerifiableCredential) {
```

Comment:
I suggest to implement this as `verifyChainOfTrustCallback(credential, verifyByRoleDefinition)` where `verifyByRoleDefinition` is block staring from line 81

Reply:
I think it is better as the current implementation, as the caller could simply pass the `credential` without thinking of the possible `callbacks` maybe? assuming the default mechanism is the RoleDefinition way.

Reply:
I mean to change only implementation of this method. Also it make sense to name it as `verifyChainOfTrustByRoleDefinitions`

Reply:
I would update the method name but I didn't really understand the implementational changes, could you please elaborate more.

Reply:
My goal is to avoid duplication in `verifyChainOfTrust` and `verifyChainOfTrustCallback`. Ideally `verifyChainOfTrustCallback` would be one line `verifyChainOfTrust(credential, this.verifyByRoleDefinition)`

Reply:
The `verifyChainOfTrustCallback` would accept external libraries for the callback. AFAIK this would lead to general purpose VC Verification something like `did-kit` `verifyCredential` method, thus the code would be different, or still I am getting confused?

Reply:
As discussed will put this aside for now